File: Sorting Searching/QuickSort.py
from random import randrange

def partition(arr, start, end):
	p_idx = randrange(start, end)
	pivot = arr[p_idx]
	arr[end], arr[p_idx] = arr[p_idx], arr[end]
	m = start
	for i in range(start, end):
		if arr[i] <= pivot:
			arr[i], arr[m] = arr[m], arr[i]
			m += 1

	arr[end], arr[m] = arr[m], arr[end]
	return m

def quick_sort(arr, start, end):
	if(start >= end):
		return
	pidx = partition(arr, start, end)
	quick_sort(arr, start, pidx-1)
	quick_sort(arr, pidx+1, end)
	return

def sort(arr):
	quick_sort(arr, 0, len(arr)-1)

li = [int(x) for x in input().split()]
sort(li)
print(li)

File: Sorting Searching/QuickSort.c
#include<stdio.h>
#include<stdlib.h>
#include<conio.h>

int partition(int arr[], int start, int end){
    int pivot = arr[end];
    int i = start - 1;
    for(int j = start; j < end; j++){
        if(arr[j] < pivot){
            i++;
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }
    int temp = arr[i + 1];
    arr[i + 1] = arr[end];
    arr[end] = temp;
    return i + 1;
}

void quickSort(int arr[], int start, int end){
    if(start < end){
        int pivot = partition(arr, start, end);
        quickSort(arr, start, pivot - 1);
        quickSort(arr, pivot + 1, end);
    }
}

int main(){
    int n;
    printf("Enter the number of elements: ");
    scanf("%d", &n);
    int *arr = (int *)malloc(n * sizeof(int));
    printf("Enter the elements: ");
    for(int i = 0; i < n; i++){
        scanf("%d", &arr[i]);
    }
    quickSort(arr, 0, n - 1);
    printf("The sorted array is: ");
    for(int i = 0; i < n; i++){
        printf("%d ", arr[i]);
    }
    free(arr);
    return 0;
}